pkg/controllers/ippool/ippool_controller.go

```diff
@@ -116,6 +122,7 @@ func (r *IPPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		log.Error(err, "unable to fetch ippool CR", "req", req.NamespacedName)
 		return resultNormal, client.IgnoreNotFound(err)
 	}
+	r.Recorder.Event(obj, v1.EventTypeNormal, "ReconcilingIPPool", "Reconciling IPPool")
```

Comment:
I think we can just post event after operation is done

Reply:
The other concern is that Reconcile call updateFail even the request will be retried. I think it should not record event if the request will be retried. since it's not the final status. The error should be divided into retriable/not retriable error.